compdb/postprocess.py

```diff
@@ -47,24 +47,20 @@
 
     local_exec_root = '__EXEC_ROOT__'
     workspace_directory = '__WORKSPACE__'
-    bazel_stderr = []
+    compile_command_json_db_files = set()
     with open(args.build_events_json_file, 'r') as f:
         for line in f:
             event = json.loads(line)
             if 'started' in event:
                 workspace_directory = event['started']['workspaceDirectory']
                 print("Workspace Directory:", workspace_directory)
-            elif 'progress' in event:
-                if 'stderr' in event['progress']:                  
-                    bazel_stderr.extend(event['progress']['stderr'].splitlines())
             elif 'workspaceInfo' in event:
                 local_exec_root = event['workspaceInfo']['localExecRoot']
                 print('Execution Root:', local_exec_root)
-
-    compile_command_json_db_files = []
-    for line in bazel_stderr:
-        if line.endswith('.compile_commands.json'):
-            compile_command_json_db_files.append(line.strip())
+            elif 'namedSetOfFiles' in event and 'files' in event['namedSetOfFiles']:
+                for file in event['namedSetOfFiles']['files']:
+                    if 'uri' in file and file['uri'].endswith('.compile_commands.json'):
+                        compile_command_json_db_files.add(file['uri'][7:])
```

Comment:
```suggestion
                        compile_command_json_db_files.add(file['uri'].strip('file://'))
```
is probably cleaner.

Reply:
Yes, recommend this change.  

In addition, please check that the file URI.startswith('file://').

If a user has gRPC `--remote_cache` enabled, these namedSetOfFiles URIs will have the form like `bytestream://remote-cache.example.com:443/blobs/26610f48167b10b99ad96c5b68570bcba259dc1c5e1adbd4f89e3130dacdc5e0/2655668`.  In that case it's impossible to know from the name alone this is a `compile_commands.json` file.